test/test_compute_service.py

```diff
@@ -272,14 +272,18 @@ def cleanup_snap(snap):
                     "vm_type")
                 vm_type = self.provider.compute.vm_types.find(
                     name=vm_type_name)[0]
-                for _ in range(vm_type.num_ephemeral_disks):
+                # Some providers, e.g. GCP, has a limit on total number of
+                # attached disks; it does not matter how many of them are
+                # ephemeral or persistent. So, wee keep in mind that we have
+                # attached 4 disks already, and add ephemeral disks accordingly
+                # to not exceed the limit.
+                for _ in range(vm_type.num_ephemeral_disks - 4):
```

Comment:
Something seems wrong here? For example, if num_ephemeral_disks were 50, subtracting 4 would mean we'd still be over the limit? It looks like we need to document that there's a disk limit that applies across the whole of cloudbridge, and maybe enforce that?